commit ddb27c8b58d28203b87aa5a7c71235338cc2fc64
Author: Scott Talbert <swt@techie.net>
Date:   Tue Mar 25 19:09:45 2014 -0400
Bug: https://github.com/signal11/hidapi/pull/168

    Add support for compiling on Debian/kFreeBSD

diff --git a/Makefile.am b/Makefile.am
index cf4f7ca..196c991 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,6 +27,10 @@ if OS_FREEBSD
 SUBDIRS += libusb
 endif
 
+if OS_KFREEBSD
+SUBDIRS += libusb
+endif
+
 if OS_WINDOWS
 SUBDIRS += windows
 endif
diff --git a/configure.ac b/configure.ac
index e5fdd4b..eb2d709 100644
--- a/configure.ac
+++ b/configure.ac
@@ -96,6 +96,17 @@ case $host in
 	AC_CHECK_LIB([iconv], [iconv_open], [LIBS_LIBUSB_PRIVATE="${LIBS_LIBUSB_PRIVATE} -liconv"], [hidapi_lib_error libiconv])
 	echo libs_priv: $LIBS_LIBUSB_PRIVATE
 	;;
+*-kfreebsd*)
+	AC_MSG_RESULT([ (kFreeBSD back-end)])
+	AC_DEFINE(OS_KFREEBSD, 1, [kFreeBSD implementation])
+	AC_SUBST(OS_KFREEBSD)
+	backend="libusb"
+	os="kfreebsd"
+	threads="pthreads"
+
+	AC_CHECK_LIB([usb], [libusb_init], [LIBS_LIBUSB_PRIVATE="${LIBS_LIBUSB_PRIVATE} -lusb"], [hidapi_lib_error libusb])
+	echo libs_priv: $LIBS_LIBUSB_PRIVATE
+	;;
 *-mingw*)
 	AC_MSG_RESULT([ (Windows back-end, using MinGW)])
 	backend="windows"
@@ -201,6 +212,7 @@ AC_SUBST([backend])
 AM_CONDITIONAL(OS_LINUX, test "x$os" = xlinux)
 AM_CONDITIONAL(OS_DARWIN, test "x$os" = xdarwin)
 AM_CONDITIONAL(OS_FREEBSD, test "x$os" = xfreebsd)
+AM_CONDITIONAL(OS_KFREEBSD, test "x$os" = xkfreebsd)
 AM_CONDITIONAL(OS_WINDOWS, test "x$os" = xwindows)
 
 AC_CONFIG_HEADERS([config.h])
diff --git a/libusb/Makefile.am b/libusb/Makefile.am
index 7f6c9dc..db65d53 100644
--- a/libusb/Makefile.am
+++ b/libusb/Makefile.am
@@ -14,6 +14,13 @@ libhidapi_la_LDFLAGS = $(LTLDFLAGS)
 libhidapi_la_LIBADD = $(LIBS_LIBUSB)
 endif
 
+if OS_KFREEBSD
+lib_LTLIBRARIES = libhidapi.la
+libhidapi_la_SOURCES = hid.c
+libhidapi_la_LDFLAGS = $(LTLDFLAGS)
+libhidapi_la_LIBADD = $(LIBS_LIBUSB)
+endif
+
 hdrdir = $(includedir)/hidapi
 hdr_HEADERS = $(top_srcdir)/hidapi/hidapi.h
 
